qiling/loader/elf.py

```diff
@@ -615,4 +615,23 @@ def load_with_ld(self, stack_addr, load_address = -1, argv = [], env = {}):
         self.new_stack = new_stack
         self.load_address = load_address
         self.images.append(self.coverage_image(load_address, load_address + mem_end, self.path))
-        self.ql.os.function_hook = FunctionHook(self.ql, self.elf_phdr + mem_start, self.elf_phnum, self.elf_phent, load_address, load_address + mem_end)
\ No newline at end of file
+        self.ql.os.function_hook = FunctionHook(self.ql, self.elf_phdr + mem_start, self.elf_phnum, self.elf_phent, load_address, load_address + mem_end)
+
+        # map vsyscall section for some specific needs
+        _vsyscall_addr = 0xffffffffff600000
+        _vsyscall_size = 0x1000
+
+        if not self.ql.mem.is_mapped(_vsyscall_addr, _vsyscall_size):
+            # initialize with \xcc then insert syscall entry
+            # each syscall should be 1KiB(0x400 bytes) away
+            self.ql.mem.map(_vsyscall_addr, _vsyscall_size, info="[vsyscall]")
+            self.ql.mem.write(_vsyscall_addr, _vsyscall_size * b'\xcc')
+
+            _vsyscall_entry_asm = [0x60c0c748, # gettimeofday
+                                   0xc9c0c748, # time
+                                   ]
+
+            for idx, val in enumerate(_vsyscall_entry_asm):
+                self.ql.mem.write(_vsyscall_addr + idx * 0x400 + (idx+0) * 4, self.ql.pack32(val))        # entry
+                self.ql.mem.write(_vsyscall_addr + idx * 0x400 + (idx+1) * 4, self.ql.pack64(0x0f000000)) # syscall
```

Comment:
any reason to fix all these address ?

Reply:
vsyscall table should be looks like this in memory
![image](https://user-images.githubusercontent.com/10076774/85376608-84f48680-b56a-11ea-86b5-df19afab7d4e.png)